### ISSUE TITLE
`zoomLevelProperty` should be discrete

### ISSUE THREAD

**Issue by arouinfar** [phetsims/geometric-optics#337]
For #237 

The `zoomLevelProperty` is type NumberPropertyIO and accepts any value within the range 0-2. However, this range shouldn’t be treated continuously and it’s easy to break the sim with the current implementation. There are discrete zoom levels, each with its own rulers.

For an example of a discrete `zoomLevelProperty` see `circuitConstructionKitDc.introScreen.model.zoomProperty`.

**Comment by pixelzoom** [comment]
The implementation of zoom in Geometric Optics bears no resemblance to CCK.   CCK is an enumeration and has only 2 zoom levels.  Geometric Optics is more general - zoom level is an integer, uses as an index into an array of scale values. 

`zoomLevelProperty` is already 'Integer', see screenshot below.  The fact that you're able to enter non-integers is a Studio problem with the new "value entry" UI, and I believe the iO team is working on adding validation.

@arouinfar let me know if I missed something here, or if you want to discuss reimplementing zoom level in Geometric Optics.

![zoomLevelProperty](https://user-images.githubusercontent.com/3046552/156019276-8f300196-b834-4839-96d9-baa8e2a866c5.jpg)

**Comment by pixelzoom** [comment]
Also worth noting is that `zoomLevelProperty` is an integer NumberProperty because that's what is required by the common-code zoom controls.  Doing what CCK is doing requires additonal code to map between integer and enumaration value, and the name `zoomLevelProperty` is then inconsistent with other sims.  Since CCK is using a non-standard approach, perhaps it should consider renaming  `zoomLevelProperty` to something non-standard that indicates that it's an enumertion.

**Comment by arouinfar** [comment]
Thanks for the clarification @pixelzoom. I think my confusion is that the zoom magnification levels are 0.25x, 0.5x, and 1x, so I missed that `zoomLevelProperty` was an integer. I recommend adding `phetioDocumentation` that connects the value of `zoomLevelProperty` to the actual zoom magnification.

**Comment by pixelzoom** [comment]
The actual scale already exists as a Property, so I've instrumented `zoomScaleProperty`:

<img width="489" alt="screenshot_1575" src="https://user-images.githubusercontent.com/3046552/156051176-98121a89-e0d0-4fa5-8bff-1f9a4ebdbb74.png">

I improved the documentation for `zoomIndexProperty`:

<img width="671" alt="screenshot_1576" src="https://user-images.githubusercontent.com/3046552/156051231-11bfefac-3598-4542-bdd8-9d3967ea7d5f.png">

What I do NOT want to do is put the actual scale values in the `zoomIndexProperty` documentation.  That would be hard-coding something that is specified elsewhere in code, and code change in the future.   So if (for some reason) the client wants to determine which `zoomIndexProperty` values correspond to which `zoomScaleProperty` values, they have to do a little work.

@arouinfar ready for review.
